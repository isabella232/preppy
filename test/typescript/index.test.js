/* global __dirname */
import pify from "pify"
import { exec } from "child_process"
import { readFile } from "fs"
import rimraf from "rimraf"
import { resolve } from "path"

import pkg from "../../package.json"

const versionString = `preppy v${pkg.version}`

const lazyExec = pify(exec)
const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)

jest.setTimeout(20000)

test("Publish Test File via Typescript", async () => {
  await lazyDelete("./dist")

  console.log(await lazyExec(
    `node ./bin/preppy --input-lib ${resolve(__dirname, "index.tsx")} --output-folder ${resolve(__dirname, "dist")}`
  ))

  const cjs = await lazyRead(resolve(__dirname, "dist/index.cjs.js"), "utf8")
  expect(cjs.replace(versionString, "VERSION_STRING")).toMatchSnapshot()

  const esm = await lazyRead(resolve(__dirname, "dist/index.esm.js"), "utf8")
  expect(esm.replace(versionString, "VERSION_STRING")).toMatchSnapshot()

  const umd = await lazyRead(resolve(__dirname, "dist/index.umd.js"), "utf8")
  expect(umd.replace(versionString, "VERSION_STRING")).toMatchSnapshot()

  const indexDef = await lazyRead(resolve(__dirname, "dist/index.d.ts"), "utf8")
  expect(indexDef).toMatchSnapshot()

  const typesDef = await lazyRead(resolve(__dirname, "dist/types.d.ts"), "utf8")
  expect(typesDef).toMatchSnapshot()
})
