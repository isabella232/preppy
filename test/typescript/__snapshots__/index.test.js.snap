// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Test File via Typescript 1`] = `
"/*! VERSION_STRING by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var _inheritsLoose = _interopDefault(require('@babel/runtime/helpers/inheritsLoose'));
require('core-js/modules/es6.object.assign');
require('core-js/modules/web.dom.iterable');
require('core-js/modules/es6.array.iterator');
require('core-js/modules/es6.string.iterator');
require('core-js/modules/es6.set');
require('core-js/modules/es7.array.includes');
require('core-js/modules/es6.promise');
var lodash = require('lodash');

console.log(\\"CherryPick Import Lodash:\\", lodash.camelCase(\\"hello world\\") === \\"helloWorld\\");
console.log(\\"Package\\", \\"preppy\\");
console.log(\\"Target\\", \\"lib\\");
console.log(\\"ES2016 Enabled:\\", Math.pow(2, 2) === 4);
new Promise(function (resolve, reject) {
  resolve(\\"resolved\\");
}).then(function (first) {
  console.log(\\"Promise:\\", first);
});
var CONSTANT = 123;
console.log(\\"Constant:\\", CONSTANT);
var myArray = [1, 2, 3];
console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
var mySet = new Set(myArray);
console.log(\\"Supports Set:\\", mySet.add(4));

(function (supportsDefault) {
  if (supportsDefault === void 0) {
    supportsDefault = true;
  }

  console.log(\\"Supports default parameters:\\", supportsDefault);
})();

var testVariable = \\"outer\\";
var _testVariable = \\"inner\\";
console.log(\\"X Value from inner scope:\\", _testVariable);
console.log(\\"X Value from outer scope:\\", testVariable);
var source = {
  first: 1,
  second: 2
};
var destructed = Object.assign({
  third: 3
}, source);
console.log(\\"Destructed:\\", destructed);
var MyClass = function () {
  function MyClass() {
    this.onClick = function () {
      console.log(\\"Clickedx\\");
    };

    console.log(\\"Called constructor\\");
    this.helper();
    this.instanceProperty = 3;
  }

  var _proto = MyClass.prototype;

  _proto.helper = function helper(x) {
    console.log(\\"Called helper: \\" + x);
  };

  return MyClass;
}();

var SecondClass = function (_MyClass) {
  _inheritsLoose(SecondClass, _MyClass);

  function SecondClass() {
    return _MyClass.call(this, 100) || this;
  }

  return SecondClass;
}(MyClass());

console.log(\\"Initialized class:\\", new MyClass());

function helper() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(PromiseHelper()).then(function ($await_1) {
      try {
        return $return(42);
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}

helper();
var setValues = function setValues(values) {
  setState({
    values: values
  }, function () {
    return true;
  });
};

exports.MyClass = MyClass;
exports.setValues = setValues;
//# sourceMappingURL=index.cjs.js.map
"
`;

exports[`Publish Test File via Typescript 2`] = `
"/*! VERSION_STRING by Sebastian Werner <s.werner@sebastian-software.de> */
import _inheritsLoose from '@babel/runtime/helpers/inheritsLoose';
import 'core-js/modules/es6.object.assign';
import 'core-js/modules/web.dom.iterable';
import 'core-js/modules/es6.array.iterator';
import 'core-js/modules/es6.string.iterator';
import 'core-js/modules/es6.set';
import 'core-js/modules/es7.array.includes';
import 'core-js/modules/es6.promise';
import { camelCase } from 'lodash';

console.log(\\"CherryPick Import Lodash:\\", camelCase(\\"hello world\\") === \\"helloWorld\\");
console.log(\\"Package\\", \\"preppy\\");
console.log(\\"Target\\", \\"lib\\");
console.log(\\"ES2016 Enabled:\\", Math.pow(2, 2) === 4);
new Promise(function (resolve, reject) {
  resolve(\\"resolved\\");
}).then(function (first) {
  console.log(\\"Promise:\\", first);
});
var CONSTANT = 123;
console.log(\\"Constant:\\", CONSTANT);
var myArray = [1, 2, 3];
console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
var mySet = new Set(myArray);
console.log(\\"Supports Set:\\", mySet.add(4));

(function (supportsDefault) {
  if (supportsDefault === void 0) {
    supportsDefault = true;
  }

  console.log(\\"Supports default parameters:\\", supportsDefault);
})();

var testVariable = \\"outer\\";
var _testVariable = \\"inner\\";
console.log(\\"X Value from inner scope:\\", _testVariable);
console.log(\\"X Value from outer scope:\\", testVariable);
var source = {
  first: 1,
  second: 2
};
var destructed = Object.assign({
  third: 3
}, source);
console.log(\\"Destructed:\\", destructed);
var MyClass = function () {
  function MyClass() {
    this.onClick = function () {
      console.log(\\"Clickedx\\");
    };

    console.log(\\"Called constructor\\");
    this.helper();
    this.instanceProperty = 3;
  }

  var _proto = MyClass.prototype;

  _proto.helper = function helper(x) {
    console.log(\\"Called helper: \\" + x);
  };

  return MyClass;
}();

var SecondClass = function (_MyClass) {
  _inheritsLoose(SecondClass, _MyClass);

  function SecondClass() {
    return _MyClass.call(this, 100) || this;
  }

  return SecondClass;
}(MyClass());

console.log(\\"Initialized class:\\", new MyClass());

function helper() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(PromiseHelper()).then(function ($await_1) {
      try {
        return $return(42);
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}

helper();
var setValues = function setValues(values) {
  setState({
    values: values
  }, function () {
    return true;
  });
};

export { MyClass, setValues };
//# sourceMappingURL=index.esm.js.map
"
`;

exports[`Publish Test File via Typescript 3`] = `
"import { FormValues } from \\"./types\\";
export declare class MyClass {
    constructor();
    onClick: () => void;
    helper(x: string): void;
}
export declare const setValues: (values: FormValues) => void;
"
`;

exports[`Publish Test File via Typescript 4`] = `
"/**
 * Values of fields in the form
 */
export interface FormValues {
    [field: string]: any;
}
"
`;
